src/dropdown.js

```diff
@@ -110,13 +110,14 @@ export default class extends Controller {
     // data-action="click->dropdown#toggle click@window->dropdown#hide keydown.up->dropdown#previousItem keydown.down->dropdown#nextItem"
     // Note: If existing actions are already specified by the user, they will be preserved and augmented without any redundancy.
 
-    const actions = this.element.dataset.action ? this.element.dataset.action.split(' ') : []
```

Comment:
Is there any reason to put events on the root element? Shouldn't they always be on the button? 🤔

Reply:
yeah, thats what i tough but i didnt wanna break the existing implementation, in case some folks use it like that in prod

Reply:
I never intended to support other cases, so I think it's fine if we simplify it to apply to just the `buttonTarget`.

Reply:
Alrigth ill make the changes and fix the specs tomorrow!

Reply:
actually, here you go!

Reply:
Thinking about this today, the majority of events might belong on the element, but the `toggle` should definitely be on the buttonTarget.

`keydown` and `click@window` probably makes more sense on the root element?

Reply:
i was thinking the same this morning! yes, gonna do that change right now. Actually, i'm also wondering if its even a good idea to auto-generate those actions? i feel like it would be more stimulus-y if we just let the dev put it wherever they want? let me know what you want, i'll make the change

Reply:
It was submitted as a PR to help remove some of the complexity of having to add lots of data-actions like the 4 on the root element and ones on the button.

It does make it more complex to maintain and less flexible so it's a tradeoff either way.